sbi/inference/posteriors/mcmc_posterior.py

```diff
@@ -245,6 +246,18 @@ def sample(
         Returns:
             Samples from posterior.
         """
+
+        try:
+            x_o_is_iid = self.potential_fn.x_is_iid
+        except AttributeError:
+            x_o_is_iid = True
+        if not x_o_is_iid:
+            warn(
+                "The default `x_o` has `x_is_iid = False`, but you are now sampling "
+                "with a batch `x` with `x_is_iid = True`. If you want to sample non-iid"
+                "`x`, please reset `x_is_iid = False` in the potential function.",
+                stacklevel=2,
+            )
```

Comment:
when does this happen? Does the user have to explicitly use the `posterior.potential_fn` object to make it happen? 
If not, i.e., if it can happen to user that constructed the posterior using `build_posterior` (maybe without knowing anything about `potentials`), then we should add more details to this warning, e.g., "by setting `posterior.potential_fn.x_is_iid=False`". 

Or am I missing something here?

Reply:
Now that we support both `sample_batched` and `sample` to work when a batch of `x` is passed, we want to warn the user to use the correct one. If `x` is batched, and the user calls `sample`, then we assume that the `x` provided is `iid`. I think the warning here is actually superfluous, as in `sbi.utils.sbiutils` we already raise a warning that if the batch dimension of `x` is greater than 1, make sure to use the correct choice of `sample` or `sample_batched` to make sure the `x` is interpreted correctly. So maybe here we remove the warning altogether?